### PR TITLE
Add torch.version.xpu

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139466
* #139258
# Motivation
We add a new attribute `torch.version.xpu` to facilitate the problem diagnosing and version control.

# Additional Context
It is aligned with `torch.version.cuda` and `torch.version.hip`.

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

File: test/test_xpu.py [hunk 1]
```diff
@@ -428,6 +430,20 @@ def test_get_arch_list(self):
         for arch in arch_list:
             self.assertTrue(arch in flags)
 
+    def test_torch_version_xpu(self):
+        self.assertTrue(len(torch.version.xpu) == 8)
+        compiler_version = int(torch.version.xpu)
+        self.assertTrue(compiler_version >= 20230000)
+        if IS_LINUX:
```
Comment (guangyey) [inline test/test_xpu.py:437]: According to @atalman 's comments, I supplemented this UT to validate the build flow as we expected.

Comment (atalman) [reply test/test_xpu.py:437]: Thank you

**Review by atalman — APPROVED** [review]
Lgtm. Please fix lint before merging

**Comment by guangyey** [comment]
> Lgtm. Please fix lint before merging

Thanks @atalman, lint is fixed.

**Review by malfet — APPROVED** [review]

**Comment by guangyey** [comment]
The UT failure is unrelated to this PR, refer to https://github.com/pytorch/pytorch/issues/140095

**Pull request closed without merging**